### ISSUE TITLE
Conditional flow for non-uploaded files

### ISSUE THREAD

**Issue by michaelclement** [department-of-veterans-affairs/va.gov-team#77121]
The goal of this ticket is to implement the conditional flow that reminds the user to attach any files that were identified as required in a previous section of the form but not immediately uploaded. ([See section 4 of this Figma](https://www.figma.com/file/UmAtr3ULQEInMXfNFwP0g0/10-10d-User-Testing-Prep?type=design&mode=design&t=VPq5NNGsbBpnVRkV-0))

Acceptance Criteria
- [x] Flow exists and matches Figma mockup
- [x] Users are prompted to upload any remaining required files needed to complete form
- [x] Files can be attached

**Comment by michaelclement** [comment]
This ticket will involve multiple stages. Setting them down here for tracking purposes.

The functionality needed:

- [ ] An overview page to prompt user if they have missing uploads
- [ ] Check for missing applicant files
- [ ] Check for missing sponsor files
- [ ] Store state in redux (this will let us track their progress as they satisfy uploads, per the green "success" alerts on the info page)
- [ ] Link to the appropriate upload screens from the main file upload information page
- [ ] Link back to the info screen once a file has been uploaded (this may not be feasible, but will research it)
- [ ] Update the information page as they upload individual files

**Comment by michaelclement** [comment]
Demonstration of current progress


[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWVuQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a8dc7331747990b74a875a3c387606765e97ea71/uploads.mov)

**Comment by michaelclement** [comment]
Some screenshots

<img width="736" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWluQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a17d18061d633ccaeedd5e295f8ee362b1fd50e3/s2.png" alt="s2.png" />

<img width="701" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeW1uQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--78882d8e95bc745ea41cf3467bd2e2521304cf20/s3.png" alt="s3.png" />

**Comment by michaelclement** [comment]
This video shows the ability to navigate to upload screens and back to the review page once a file is uploaded.

**NOTE** The link that takes the user back to the review screen is necessary because overriding the continue button isn't currently feasible. Need to have a discussion with designers on a proper way to deal with the issue.


[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMlN1QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7f5cb9e8b3639d67befd15db51a7cbc08072adf2/upload.mov)

**Comment by michaelclement** [comment]
Currently the branch has a lot of changes, so now I'm working to break it into multiple PRs and get those approved/merged into staging. [The first is here](https://github.com/department-of-veterans-affairs/vets-website/pull/28362).

Waiting on:
- [ ] Team approval
- [ ] Platform approval

**Comment by michaelclement** [comment]
Second PR is still in the works.

**Comment by michaelclement** [comment]
Overall progress
- Still need to implement conditional warnings on the screen with edit links
- Need to break this PR into smaller ones.
 
Sub PRs:
- [Sub PR #1](https://github.com/department-of-veterans-affairs/vets-website/pull/28362) (merged)
- [Sub PR #2](https://github.com/department-of-veterans-affairs/vets-website/pull/28400) (waiting on review)
- Sub PR #3 (Drafted)

**Comment by michaelclement** [comment]
[Final PR was merged](https://github.com/department-of-veterans-affairs/vets-website/pull/28312). @baltes this ticket is ready to close whenever you've reviewed it.
